### PR TITLE
fix directory upload to deliver expected directory structure

### PR DESCRIPTION
**Pull request by pdxwebdev**
This fix comes from this use case:
1. I want to upload a directory
2. I upload a director with the path /home/username/somedir

Current behavior:
The skylink directory structure will be SKYLINK_HASH/home/username/somedir/index.html

New behavior:
The skylink directory structure will be SKYLINK_HASH/index.html

### CODE REVIEW

File: siaskynet/skynet.py [hunk 1]
```diff
@@ -76,10 +76,10 @@ def upload_directory_request(path, opts=None):
             opts = Skynet.default_upload_options()
 
         ftuples = []
-        files = list(Skynet.walk_directory(path).keys())
+        files = list(Skynet.walk_directory(path, path).keys())
         for file in files:
             ftuples.append((opts.portal_directory_file_fieldname,
-                            (file, open(file, 'rb'))))
+                            (file, open(path + file, 'rb'))))
```
Comment (peterjanbrone) [inline siaskynet/skynet.py]: Could we have just done the following? 
```
 ftuples.append((opts.portal_directory_file_fieldname, (file[len(path):], open(file, 'rb'))))
```

Comment (pdxwebdev) [reply siaskynet/skynet.py]: Yes, great suggestion. I've made the change.

**Comment by pdxwebdev** [comment]
@peterjanbrone I made the requested change. PR is ready to go. Feel free to delete the other PR I already closed. It was created in error. Thanks.

File: siaskynet/skynet.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def upload_directory_request(path, opts=None):
         files = list(Skynet.walk_directory(path).keys())
         for file in files:
             ftuples.append((opts.portal_directory_file_fieldname,
-                            (file, open(file, 'rb'))))
+                            (file[len(path):], open(file, 'rb'))))
```
Comment (m-cat) [inline siaskynet/skynet.py:82]: Shouldn't it be `[:len(path)]` if you want to cut the initial path? Also, I would first assert that `file` starts with `path`.

**Comment by peterjanbrone** [comment]
@pdxwebdev  sorry this took so long to merge - our priorities shifted for a moment there but we're very much focussing our attention to the SDKs at the moment 💪  thank you for contributing

**Pull request merged**